actors/builtin/market/market_test.go

```diff
@@ -1 +1,252 @@
 package market_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/filecoin-project/specs-actors/actors/abi"
+	"github.com/filecoin-project/specs-actors/actors/abi/big"
+	"github.com/filecoin-project/specs-actors/actors/builtin"
+	"github.com/filecoin-project/specs-actors/actors/builtin/market"
+	"github.com/filecoin-project/specs-actors/actors/builtin/miner"
+	"github.com/filecoin-project/specs-actors/actors/runtime/exitcode"
+	"github.com/filecoin-project/specs-actors/actors/util/adt"
+	"github.com/filecoin-project/specs-actors/support/mock"
+	tutil "github.com/filecoin-project/specs-actors/support/testing"
+)
+
+func TestActorConstruction(t *testing.T) {
+	actor := market.Actor{}
+	receiver := tutil.NewIDAddr(t, 100)
+	builder := mock.NewBuilder(context.Background(), receiver).
+		WithCaller(builtin.SystemActorAddr, builtin.InitActorCodeID)
+
+	t.Run("simple construction", func(t *testing.T) {
+		rt := builder.Build(t)
+
+		rt.ExpectValidateCallerAddr(builtin.SystemActorAddr)
+
+		params := adt.EmptyValue{}
+		ret := rt.Call(actor.Constructor, &params).(*adt.EmptyValue)
+
+		assert.Equal(t, adt.EmptyValue{}, *ret)
+		rt.Verify()
+	})
+}
+
+func TestAddBalance(t *testing.T) {
+	marketActor := tutil.NewIDAddr(t, 100)
+	provider := tutil.NewIDAddr(t, 101)
+	worker := tutil.NewIDAddr(t, 102)
+	client := tutil.NewIDAddr(t, 103)
+
+	var st market.State
+
+	setup := func() (*mock.Runtime, *marketActorTestHarness) {
+		builder := mock.NewBuilder(context.Background(), marketActor).
+			WithBalance(abi.NewTokenAmount(100), abi.NewTokenAmount(0)).
```

Comment:
I'd prefer not to set the balance inaccurately here. Set it only directly when it should known. For example, it would be pretty reasonable for the actor to have a sanity check that the sum of its balance tables matches the balance actually held  (I'd love you to add a facility like that to the state object, actually).

I suspect right now the actor never looks at it's balance so we could get away with zero always until we add such checks.